### PR TITLE
added renderlabel function and a run data paragraph with it

### PR DESCRIPTION
**Pull request by msmner**

### CODE REVIEW

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx [hunk 1]
```diff
@@ -35,7 +36,7 @@ const TestRunDetails = ({ testRun }: ITestRunDetailsProps) => {
         ),
         [ testRun ],
     );
-
+
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:39]: remove spacing

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx [hunk 2]
```diff
@@ -52,7 +53,29 @@ const TestRunDetails = ({ testRun }: ITestRunDetailsProps) => {
         return testRunText;
     }, [ testRun ]);
 
-    const renderTimeAndMemoryUsed = useCallback(() => (
+    const renderLabel = useCallback(
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:56]: I think renderResultTypeLabel  naming will indicate better what the label is for

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx [hunk 3]
```diff
@@ -52,7 +53,29 @@ const TestRunDetails = ({ testRun }: ITestRunDetailsProps) => {
...
+        () => {
+            const result = testRun.resultType.toLowerCase();
+            const type = result === 'correctanswer'
+                ? LabelType.success
+                : result === 'wronganswer'
+                    ? LabelType.danger
+                    : LabelType.warning;
+            
+            const resultSplit = testRun.resultType.split(/(?=[A-Z])/).join(' ');
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:65]: Move this into util function

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx [hunk 4]
```diff
@@ -52,7 +53,29 @@ const TestRunDetails = ({ testRun }: ITestRunDetailsProps) => {
         return testRunText;
     }, [ testRun ]);
 
-    const renderTimeAndMemoryUsed = useCallback(() => (
+    const renderLabel = useCallback(
+        () => {
+            const result = testRun.resultType.toLowerCase();
+            const type = result === 'correctanswer'
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:59]: Could be a function

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx [hunk 5]
```diff
@@ -52,7 +53,29 @@ const TestRunDetails = ({ testRun }: ITestRunDetailsProps) => {
...
+                ? LabelType.success
+                : result === 'wronganswer'
+                    ? LabelType.danger
+                    : LabelType.warning;
+            
+            const resultSplit = testRun.resultType.split(/(?=[A-Z])/).join(' ');
+
+            return (
+                <Label
+                        type={type}
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:69]: format here

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx [hunk 6]
```diff
@@ -129,7 +154,7 @@ const TestRunDetails = ({ testRun }: ITestRunDetailsProps) => {
         testRun,
         user.permissions.canAccessAdministration,
     ]);
-
+
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:157]: unnesessary spaces here

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx [hunk 7]
```diff
@@ -69,11 +92,13 @@ const TestRunDetails = ({ testRun }: ITestRunDetailsProps) => {
                 />
                 <span>
                     {testRun.memoryUsed}
-                    MB
                 </span>
             </span>
+            <span className={styles.testRunDataParagraph}>
+                {renderLabel()}
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:98]: spaces after {    and     }

**Review by shsimeonova — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx [hunk 8]
```diff
@@ -52,7 +53,29 @@ const TestRunDetails = ({ testRun }: ITestRunDetailsProps) => {
         return testRunText;
     }, [ testRun ]);
 
-    const renderTimeAndMemoryUsed = useCallback(() => (
+    const renderLabel = useCallback(
+        () => {
+            const result = testRun.resultType.toLowerCase();
+            const type = result === 'correctanswer'
+                ? LabelType.success
+                : result === 'wronganswer'
```
Comment (mishogv) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:61]: 'wronganswer' can be extracted as constant

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx (hunk 4 above)
Comment (mishogv) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:59]: 'correctanswer' can be extracted as constant

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx (hunk 1 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:39]: Done.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx (hunk 2 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:56]: Renamed.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx (hunk 3 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:65]: Done.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx (hunk 8 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:61]: Done.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx (hunk 6 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:157]: ESlint took care of it.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx (hunk 5 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:69]: Done.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx (hunk 4 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:59]: Done.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx (hunk 7 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/test-run-details/TestRunDetails.tsx:98]: Not sure what you meant. I hope ESlint fixed it though.

**Review by shsimeonova — CHANGES_REQUESTED** [review]

**Review by shsimeonova — APPROVED** [review]

**Review by shsimeonova — APPROVED** [review]

**Review by mishogv — APPROVED** [review]

**Pull request merged**